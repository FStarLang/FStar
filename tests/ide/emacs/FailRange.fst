(*
   Copyright 2008-2018 Microsoft Research

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*)
module FailRange
open FStar.Tactics

let a : unit =
  assert_by_tactic (True /\ True)
    (fun () -> split (); fail "A")

let b () : unit =
  assert_by_tactic (True /\ True)
    (fun () -> split (); fail "B")

let c : unit =
  assert_by_tactic (True /\ True)
    (fun () -> fail "C")

let d () : unit =
  assert_by_tactic (True /\ True)
    (fun () -> fail "D")
