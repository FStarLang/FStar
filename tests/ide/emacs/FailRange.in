{"query-id":"1","query":"push","args":{"kind":"full","code":"module FailRange\nopen FStar.Tactics\n","line":1,"column":0}}
{"query-id":"2","query":"push","args":{"kind":"full","code":"\nlet a : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> split (); fail \"A\")\n","line":3,"column":0}}
{"query-id":"3","query":"push","args":{"kind":"lax","code":"\nlet a : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> split (); fail \"A\")\n","line":3,"column":0}}
{"query-id":"4","query":"push","args":{"kind":"full","code":"\nlet b () : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> split (); fail \"B\")\n","line":7,"column":0}}
{"query-id":"5","query":"push","args":{"kind":"lax","code":"\nlet b () : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> split (); fail \"B\")\n","line":7,"column":0}}
{"query-id":"6","query":"push","args":{"kind":"full","code":"\nlet c : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> fail \"C\")\n","line":11,"column":0}}
{"query-id":"7","query":"push","args":{"kind":"lax","code":"\nlet c : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> fail \"C\")\n","line":11,"column":0}}
{"query-id":"8","query":"push","args":{"kind":"full","code":"\nlet d () : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> fail \"D\")\n","line":15,"column":0}}
{"query-id":"9","query":"push","args":{"kind":"lax","code":"\nlet d () : unit =\n  assert_by_tactic (True /\\ True)\n    (fun () -> fail \"D\")\n","line":15,"column":0}}
