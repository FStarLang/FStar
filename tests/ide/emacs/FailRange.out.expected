{"kind": "protocol-info", "rest": "[...]"}
{"kind": "response", "query-id": "1", "response": [], "status": "success"}
{"contents": {"depth": 0, "goals": [{"goal": {"type": "squash l_True", "witness": "(*?u[...]*) _"}, "hyps": []}, {"goal": {"type": "squash l_True", "witness": "(*?u[...]*) _"}, "hyps": []}], "label": "at the time of failure", "location": "<location removed>", "smt-goals": []}, "kind": "message", "level": "proof-state", "query-id": "2"}
{"kind": "response", "query-id": "2", "response": [{"level": "error", "message": "user tactic failed: A", "ranges": [{"beg": [5, 2], "end": [5, 18], "fname": "<input>"}, {"beg": [6, 25], "end": [6, 33], "fname": "<input>"}]}], "status": "failure"}
{"kind": "response", "query-id": "3", "response": [], "status": "success"}
{"contents": {"depth": 0, "goals": [{"goal": {"type": "squash l_True", "witness": "(*?u[...]*) _"}, "hyps": [{"name": "uu___", "type": "unit"}]}, {"goal": {"type": "squash l_True", "witness": "(*?u[...]*) _"}, "hyps": [{"name": "uu___", "type": "unit"}]}], "label": "at the time of failure", "location": "<location removed>", "smt-goals": []}, "kind": "message", "level": "proof-state", "query-id": "4"}
{"kind": "response", "query-id": "4", "response": [{"level": "error", "message": "user tactic failed: B", "ranges": [{"beg": [8, 11], "end": [9, 2], "fname": "<input>"}, {"beg": [10, 25], "end": [10, 33], "fname": "<input>"}]}], "status": "failure"}
{"kind": "response", "query-id": "5", "response": [], "status": "success"}
{"contents": {"depth": 0, "goals": [{"goal": {"type": "squash (l_True /\\ l_True)", "witness": "(*?u[...]*) _"}, "hyps": []}], "label": "at the time of failure", "location": "<location removed>", "smt-goals": []}, "kind": "message", "level": "proof-state", "query-id": "6"}
{"kind": "response", "query-id": "6", "response": [{"level": "error", "message": "user tactic failed: C", "ranges": [{"beg": [13, 2], "end": [13, 18], "fname": "<input>"}]}], "status": "failure"}
{"kind": "response", "query-id": "7", "response": [], "status": "success"}
{"contents": {"depth": 0, "goals": [{"goal": {"type": "squash (l_True /\\ l_True)", "witness": "(*?u[...]*) _"}, "hyps": [{"name": "uu___", "type": "unit"}]}], "label": "at the time of failure", "location": "<location removed>", "smt-goals": []}, "kind": "message", "level": "proof-state", "query-id": "8"}
{"kind": "response", "query-id": "8", "response": [{"level": "error", "message": "user tactic failed: D", "ranges": [{"beg": [16, 11], "end": [17, 2], "fname": "<input>"}]}], "status": "failure"}
{"kind": "response", "query-id": "9", "response": [], "status": "success"}
